extension/storage/dbstorage/client.go

```diff
@@ -59,45 +59,37 @@ func newClient(ctx context.Context, driverName string, db *sql.DB, tableName str
 
 // Get will retrieve data from storage that corresponds to the specified key
 func (c *dbStorageClient) Get(ctx context.Context, key string) ([]byte, error) {
-	rows, err := c.getQuery.QueryContext(ctx, key)
-	if err != nil {
-		return nil, err
-	}
-	if !rows.Next() {
-		return nil, nil
-	}
-	var result []byte
-	err = rows.Scan(&result)
-	if err != nil {
-		return result, err
-	}
-	err = rows.Close()
-	return result, err
+	return c.get(ctx, key, nil)
 }
 
 // Set will store data. The data can be retrieved using the same key
 func (c *dbStorageClient) Set(ctx context.Context, key string, value []byte) error {
-	_, err := c.setQuery.ExecContext(ctx, key, value, value)
-	return err
+	return c.set(ctx, key, value, nil)
 }
 
 // Delete will delete data associated with the specified key
 func (c *dbStorageClient) Delete(ctx context.Context, key string) error {
-	_, err := c.deleteQuery.ExecContext(ctx, key)
-	return err
+	return c.delete(ctx, key, nil)
 }
 
 // Batch executes the specified operations in order. Get operation results are updated in place
 func (c *dbStorageClient) Batch(ctx context.Context, ops ...*storage.Operation) error {
-	var err error
+	// Start a new transaction
+	tx, err := c.db.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	//nolint:errcheck
```

Comment:
do we not want to do something on error here?

Reply:
Added logging here
Unfortunately there was no logger instance in this object, so have to inject it, I believe it might be useful in this object for future improvements